pkg/test/framework/components/echo/echotest/setup.go

```diff
@@ -42,13 +42,21 @@ func (t *T) Setup(setupFn srcSetupFn) *T {
 }
 
 func (t *T) setup(ctx framework.TestContext, srcInstances echo.Callers) {
+	if !t.hasSourceSetup() {
```

Comment:
For the source setup, I'm not sure if we need to also check destination setup? If we had destination/pair level setup we'd get dumps for each leaf but not the outer scope.